core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -370,8 +376,30 @@ trait WorkspaceComponent {
       filter(rec => (rec.namespace === namespaceName.value))
     }
 
+    def listOutdatedEntityCaches(limit: Int): ReadAction[Seq[(UUID, Timestamp)]] = {
+      //We don't necessarily want all of the workspaces back, just the ones that we can handle in one pass
+      //We also order by lastModified so we can handle the ones that are the longest out of date
+      filter(rec => (rec.entityCacheLastUpdated < rec.lastModified)).take(limit).sortBy(_.lastModified.asc).map { ws => (ws.id, ws.lastModified) }.result
+    }
+
+    def listBackloggedEntityCaches(limit: Int): ReadAction[Seq[(UUID, Timestamp)]] = {
+      filter(rec => rec.entityCacheLastUpdated.isEmpty).take(limit).map { ws => (ws.id, ws.lastModified) }.result
+    }
```

Comment:
add `.sortBy(_.lastModified.desc)`, so we're catching up on the most-recently updated workspaces first?